scene/main/node.cpp

```diff
@@ -417,15 +417,18 @@ void Node::_propagate_groups_dirty() {
 }
 
 void Node::add_child_notify(Node *p_child) {
-	// to be used when not wanted
+	// Called when a node is added as a child to this one.
+	GDVIRTUAL_CALL(_add_child_notify, p_child);
 }
 
 void Node::remove_child_notify(Node *p_child) {
-	// to be used when not wanted
+	// Called when a child is removed from this node.
+	GDVIRTUAL_CALL(_remove_child_notify, p_child);
 }
 
 void Node::move_child_notify(Node *p_child) {
-	// to be used when not wanted
+	// Called when a child is moved inside this node.
+	GDVIRTUAL_CALL(_move_child_notify, p_child);
 }
 
 void Node::owner_changed_notify() {
```

Comment:
This was added recently, but for consistency I guess it could also get a `GDVIRTUAL_CALL` (and call to base class in Light2D and Light3D where it's overridden).

Reply:
I can't seem to see where its overriden in Light2D and Light3D, I do see the add and remove notifies overriden in Window without making a base call, I'm supposing that needs to be corrected here as well. TabContainer is also unique in that the TabBar seems to be hidden as an internal node from this method, I believe all the other internal nodes will get notified by this since none of their parents' overrides the notify, unless the notify isn't being called on the other internal nodes for some reason. Probably could fix the internal TabBar being hidden and document that all internal nodes will be shown by this method if that's the case.

Reply:
Actually I believe this is somewhat unique compared to the signal in that it gets called regardless of whether its in the tree or not? The `child_enter_tree` and `child_exit_tree` are only supposed to run when entering and exiting a scene tree since they propagate whereas this runs on every `_add_child_nocheck`, `move_child`, and `remove_child` call.

Reply:
To be clear, I was commenting about `owner_changed_notify`:
```
$ rg owner_changed_notify
scene/3d/light_3d.h
88:     virtual void owner_changed_notify() override;

scene/3d/light_3d.cpp
179:void Light3D::owner_changed_notify() {

scene/main/node.h
212:    virtual void owner_changed_notify();

scene/main/node.cpp
431:void Node::owner_changed_notify() {
1559:   owner_changed_notify();

scene/2d/light_2d.cpp
33:void Light2D::owner_changed_notify() {

scene/2d/light_2d.h
76:     virtual void owner_changed_notify() override;
```

But it's good that it led you to double check code and find another child notify override in `Window` :)